addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportApi.java

```diff
@@ -0,0 +1,318 @@
+                        }
+                        reportData.setIncludeConfidence(confidenceIndex, true);
+                    }
+                } else {
+                    reportData.setIncludeAllConfidences(true);
+                }
+
+                // Risks
+                if (isContainedParam(PARAM_INC_RISKS, params)) {
+                    reportData.setIncludeAllRisks(false);
+                    String[] risks = params.getString(PARAM_INC_RISKS).split(DELIMITER_REGEX);
+                    for (String risk : risks) {
+                        risk = StringUtils.capitalize(risk.trim().toLowerCase(Locale.ROOT));
+                        int riskIndex = ArrayUtils.indexOf(Alert.MSG_RISK, risk);
+                        if (riskIndex == -1) {
+                            throw new ApiException(Type.ILLEGAL_PARAMETER, PARAM_INC_RISKS);
+                        }
+                        reportData.setIncludeRisk(riskIndex, true);
+                    }
+                } else {
+                    reportData.setIncludeAllRisks(true);
+                }
+
+                // Alerts
+                reportData.setAlertTreeRootNode(extReports.getFilteredAlertTree(reportData));
+
+                try {
+                    extReports.generateReport(reportData, template, reportFilePath, display);
+                } catch (Exception e) {
+                    throw new ApiException(Type.INTERNAL_ERROR, e);
+                }
+                return new ApiResponseElement("generatedReportPath", reportFilePath);
+            default:
+                throw new ApiException(Type.BAD_ACTION);
+        }
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        LOG.debug("Request for handleApiAction: {} (params: {})", name, params);
+        switch (name) {
+            case VIEW_TEMPLATES:
+                ApiResponseList resultList = new ApiResponseList(name);
+                extReports
+                        .getTemplates()
+                        .forEach(
+                                t ->
+                                        resultList.addItem(
+                                                new ApiResponseElement(
+                                                        "template", t.getConfigName())));
+                return resultList;
+            case VIEW_TEMPLATE_DETAILS:
+                Template template =
+                        extReports.getTemplateByConfigName(params.getString(PARAM_TEMPLATE));
+                if (template == null) {
+                    throw new ApiException(
+                            Type.DOES_NOT_EXIST,
+                            Constant.messages.getString(
+                                    "reports.api.error.templateDoesNotExist",
+                                    params.getString(PARAM_TEMPLATE)));
+                }
+                ApiResponseSet<Object> resultSet = new ApiResponseSet<>(name, new HashMap<>());
```

Comment:
The XML response requires parsing the fields themes and sections to access the values.